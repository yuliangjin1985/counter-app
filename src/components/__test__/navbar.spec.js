import React from "react";
import { shallow } from "enzyme";

import NavBar from "../navbar.jsx";

describe("NavBar", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("test for navbar", () => {
    const len = wrapper.find("nav").length;
    expect(len).toBe(1);
    expect(wrapper.find("Link").length).toBe(4);
  });
});
