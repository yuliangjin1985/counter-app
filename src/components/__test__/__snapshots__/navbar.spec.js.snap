// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/news"
          >
            News
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/calculator"
          >
            Calculator
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/counter"
          >
            Counter
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/todos"
          >
            Todos
          </Link>
        </div>
      </div>,
      "className": "navbar navbar-expand-lg navbar-light bg-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="navbar-nav"
        >
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/news"
          >
            News
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/calculator"
          >
            Calculator
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/counter"
          >
            Counter
          </Link>
          <Link
            className="nav-item nav-link active"
            replace={false}
            to="/todos"
          >
            Todos
          </Link>
        </div>,
        "className": "collapse navbar-collapse",
        "id": "navbarNavAltMarkup",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/news"
            >
              News
            </Link>,
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/calculator"
            >
              Calculator
            </Link>,
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/counter"
            >
              Counter
            </Link>,
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/todos"
            >
              Todos
            </Link>,
          ],
          "className": "navbar-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "News",
              "className": "nav-item nav-link active",
              "replace": false,
              "to": "/news",
            },
            "ref": null,
            "rendered": "News",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Calculator",
              "className": "nav-item nav-link active",
              "replace": false,
              "to": "/calculator",
            },
            "ref": null,
            "rendered": "Calculator",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Counter",
              "className": "nav-item nav-link active",
              "replace": false,
              "to": "/counter",
            },
            "ref": null,
            "rendered": "Counter",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Todos",
              "className": "nav-item nav-link active",
              "replace": false,
              "to": "/todos",
            },
            "ref": null,
            "rendered": "Todos",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/news"
            >
              News
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/calculator"
            >
              Calculator
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/counter"
            >
              Counter
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/todos"
            >
              Todos
            </Link>
          </div>
        </div>,
        "className": "navbar navbar-expand-lg navbar-light bg-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="navbar-nav"
          >
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/news"
            >
              News
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/calculator"
            >
              Calculator
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/counter"
            >
              Counter
            </Link>
            <Link
              className="nav-item nav-link active"
              replace={false}
              to="/todos"
            >
              Todos
            </Link>
          </div>,
          "className": "collapse navbar-collapse",
          "id": "navbarNavAltMarkup",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="nav-item nav-link active"
                replace={false}
                to="/news"
              >
                News
              </Link>,
              <Link
                className="nav-item nav-link active"
                replace={false}
                to="/calculator"
              >
                Calculator
              </Link>,
              <Link
                className="nav-item nav-link active"
                replace={false}
                to="/counter"
              >
                Counter
              </Link>,
              <Link
                className="nav-item nav-link active"
                replace={false}
                to="/todos"
              >
                Todos
              </Link>,
            ],
            "className": "navbar-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "News",
                "className": "nav-item nav-link active",
                "replace": false,
                "to": "/news",
              },
              "ref": null,
              "rendered": "News",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Calculator",
                "className": "nav-item nav-link active",
                "replace": false,
                "to": "/calculator",
              },
              "ref": null,
              "rendered": "Calculator",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Counter",
                "className": "nav-item nav-link active",
                "replace": false,
                "to": "/counter",
              },
              "ref": null,
              "rendered": "Counter",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Todos",
                "className": "nav-item nav-link active",
                "replace": false,
                "to": "/todos",
              },
              "ref": null,
              "rendered": "Todos",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
